tools/android/MASTG-TOOL-0111.md

```diff
@@ -0,0 +1,18 @@
+---
+title: Blutter
+platform: android
+source: https://github.com/worawit/blutter
+---
+
+[Blutter](https://github.com/worawit/blutter) is an open-source tool created to support the reverse engineering of Flutter applications by compiling the Dart AOT Runtime. It targets the lib files found in decompiled apks and is compatible with the latest versions of Dart. It makes use of an advanced C++20 formatting library. The tool is compatible with Linux, Windows, and macOS operating systems. Blutter is capable of extracting and analyzing Dart objects and can generate Frida scripts for further analysis. It automatically compiles any required Dart versions that are not already installed.
+
+This tool does require a specific environment to work, which can be found [here](https://github.com/worawit/blutter?tab=readme-ov-file#environment-setup).If you don’t want to setup the environment here is the docker support for the blutter tool.
+
+Use the apktool to Extract "lib" directory from apk file.
```

Comment:
Update this to either show both options: either apktool d yourself first, or directly supply the apk file to blutter.

Reply:
should this go into a technique?

Reply:
A technique might not be needed as the tool usage and its output details are given [here](https://github.com/OWASP/owasp-mastg/blob/master/tools/android/MASTG-TOOL-0011.md). However, I do see that the lib folder details are missing.

Reply:
@ScreaMy7 When we add a tool, it usually needs to have one or more MASTG-TECH-xxxx that use that tool (https://mas.owasp.org/MASTG/techniques/). This was a requirement in the past, and we need to require it again because the tools section is not meant to be "just" a list of tools. We need to ensure that whoever adds a tool has tested it and demonstrated that it is actually useful, and the way to ensure this is to create techniques for the most useful use cases of the tool. Then we'd have to make sure those techniques are used in testing.

From time to time, we review tools, and if there are no use cases for the tools in the MASTG, we remove them.

In this case we'd need a MASTG-TECH-00xx like "Reverse Engineering Flutter Apps for Android".

Reply:
A technique and its usage are given by the datafarm-research team in #2913